Completed main tests

Rafsan7238 opened the pull request:
Reached 100% main test coverage.

Rafsan7238 committed:
Added rest of the dfam file tests

Rafsan7238 committed:
Updated coding to utf-8 for compatibility

Rafsan7238 committed:
Changed the coding back to future_typing

Rafsan7238 committed:
Updated python version to >=3.10

Rafsan7238 committed:
Removed future_typing

Rafsan7238 committed:
Update pyproject.toml

Rafsan7238 committed:
Update pyproject.toml

Rafsan7238 committed:
Update pyproject.toml

Rafsan7238 committed:
Regenerated poetry lock

Rafsan7238 committed:
Updated coverage and pytest versions

Rafsan7238 committed:
Added rest of the io tests

Rafsan7238 committed:
Added test_refseq

Rafsan7238 committed:
Added test for download_dfam

rbturnbull commented on tests/test_main.py:
```diff
@@ -95,7 +99,10 @@ def test_add_fail():
 
 
 def copy_NC_036113(url, path):
```
This function should have a different name now.

Rafsan7238 replied:
It has been done now.

rbturnbull reviewed (changes requested):
Can you please change the name of the function`copy_NC_036113` and make it more general? maybe just 'copy_dummy_fasta'

Rafsan7238 committed:
Update test_main.py

Integrated code review suggestion; replaced function copy_NC_036113 with copy_dummy_fasta

Rafsan7238 commented:
Incorporated the requested function name change. Since the test cases were approved before, merging the branch.
